### PR TITLE
Make heap growth more aggresive

### PR DESCRIPTION
**Pull request by densh**
Current heuristic often ends up stabilizing upon insufficient heap
size (i.e. we can get a speedup by getting a bigger heap.)

Hereby we change it to grow at 1.41x increments, until at least
half the blocks are free after every collection. This largely
compensates for the fact that we don't have young generation.

Moreover, before the heap growth to 128MB we grow even faster at 2x
increments. This minimizes overhead of apps that allocate a lot early
on during their initialization process. 

Additionally, we also increase minimal heap size to 4MB instead of 1MB. 

Review @LukasKellenberger

### CODE REVIEW

**Comment by densh** [comment]
**Note:** Previously we would require a 1/3 of blocks to be free and grow in 1.3 increments.

File: nativelib/src/main/resources/gc/immix/Constants.h [hunk 1]
```diff
@@ -49,6 +49,8 @@
 
 #define LARGE_BLOCK_MASK (~((1 << LARGE_OBJECT_MIN_SIZE_BITS) - 1))
 
-#define GROWTH_RATE 30
+#define EARLY_GROWTH_THRESHOLD (128 * 1024 * 1024UL)
+#define EARLY_GROWTH_RATE 2.0
+#define GROWTH_RATE 1.414213562
```
Comment (LukasKellenberger) [inline nativelib/src/main/resources/gc/immix/Constants.h]: Is there a reason for using sqrt(2) and not 1.4 ?

Comment (densh) [reply nativelib/src/main/resources/gc/immix/Constants.h]: sqrt(2) makes heap double (roughly) every 2 heap size extensions. With 1.4 it's going to be a bit less than that.

File: nativelib/src/main/resources/gc/immix/ImmixGC.c [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include "State.h"
 #include "utils/MathUtils.h"
 
-#define INITIAL_HEAP_SIZE (1024 * 1024UL)
+#define INITIAL_HEAP_SIZE (4 * 1024 * 1024UL)
```
[the remark this answers is not in the document]
Comment (densh) [reply nativelib/src/main/resources/gc/immix/ImmixGC.c]: Excellent point, lets keep 1MB for large heap for now.

[the remark this answers is not in the document]
Comment (densh) [reply nativelib/src/main/resources/gc/immix/ImmixGC.c]: 2aae100 addresses this by introducing a separate constant for initial large heap size.

**Pull request merged**